<a href="https://colab.research.google.com/github/VitorBD32/ATIVIDADE-USANDO-GITHUB/blob/main/tcers_extra%C3%A7%C3%A3o%20de%20dados.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [16]:
# Search for any names containing 'Vitor' (case-insensitive) in the cleaned DataFrame
similar_names_vitor = candidates_df_cleaned_names[candidates_df_cleaned_names['Candidate Name'].str.contains('Vitor', na=False, case=False)]

if not similar_names_vitor.empty:
    print("Nomes de candidatos contendo 'Vitor':")
    print(similar_names_vitor[['Candidate ID', 'Candidate Name', 'P1 Score', 'P2 Score', 'Total Score', 'Eliminated', 'Elimination Reason']].to_string(index=False))
else:
    print("Nenhum candidato com 'Vitor' no nome encontrado no DataFrame.")

Nomes de candidatos contendo 'Vitor':
Candidate ID                              Candidate Name  P1 Score  P2 Score  Total Score  Eliminated                      Elimination Reason
    10013297                  Alexsandro Montiel Vitoria      40.0      38.0         78.0       False                                        
    10011048                   Ananias Vitor Luis Santos      32.0      22.0         54.0       False                                        
    10000689        Andressa Vitoria Lapazin de Oliveira      26.0      17.0         43.0       False                                        
    10001030        Carolyn Vitorya da Silveira Jacobsen      13.0       8.0         21.0        True Failed minimum score requirements (OCE)
    10015218             Eduarda Vitoria Valente Moreira      14.0      24.0         38.0       False                                        
    10019569                      Emily Vitoria Uflacker      27.0      17.0         44.0       False         

In [17]:
candidate_name_to_find = "Vitor de Brito Cardoso Oliveira"

# Find the candidate in the cleaned DataFrame
specific_vitor_candidate = candidates_df_cleaned_names[candidates_df_cleaned_names['Candidate Name'] == candidate_name_to_find]

if not specific_vitor_candidate.empty:
    print(f"Detalhes do candidato '{candidate_name_to_find}':")
    display(specific_vitor_candidate[['Candidate ID', 'Candidate Name', 'P1 Score', 'P2 Score', 'Total Score', 'Eliminated', 'Elimination Reason']])

    # Check if the candidate is in the non_eliminated_candidates_df
    vitor_non_eliminated = non_eliminated_candidates_df[non_eliminated_candidates_df['Candidate ID'] == specific_vitor_candidate['Candidate ID'].iloc[0]]

    if not vitor_non_eliminated.empty:
        # Find the rank of the candidate among non-eliminated candidates
        # The non_eliminated_candidates_df is already sorted by 'Total Score' in descending order
        # Reset index to get a continuous rank
        ranked_df = non_eliminated_candidates_df.reset_index(drop=True)
        rank = ranked_df[ranked_df['Candidate ID'] == specific_vitor_candidate['Candidate ID'].iloc[0]].index[0] + 1
        print(f"\nO candidato {candidate_name_to_find} está classificado na posição: {rank} (de {len(non_eliminated_candidates_df)} candidatos não eliminados).")
    else:
        print(f"\nO candidato {candidate_name_to_find} foi eliminado com base nos critérios de pontuação.")
else:
    print(f"Candidato '{candidate_name_to_find}' não encontrado no DataFrame após busca exata.")


Candidato 'Vitor de Brito Cardoso Oliveira' não encontrado no DataFrame após busca exata.


# Task
Extract elimination criteria from the `Ed_1_TCE_RS_24_abertura.pdf` file and explain that candidate data is needed to apply these criteria and count non-eliminated candidates.

## Extract Text from Edital PDF

### Subtask:
Extract all textual content from the provided edital PDF (`Ed_1_TCE_RS_24_abertura.pdf`) to make it searchable and parsable.


**Reasoning**:
The 'pypdf' library is required to extract text from PDF files. This step installs the library to make it available for use.



In [1]:
pip install pypdf

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 329.5/329.5 kB 6.4 MB/s eta 0:00:00


**Reasoning**:
Now that 'pypdf' is installed, I will use it to extract all text content from the specified PDF file, iterating through each page and concatenating the extracted text.



In [2]:
from pypdf import PdfReader

pdf_path = '/content/Ed_1_TCE_RS_24_abertura.pdf'

text_content = ''
reader = PdfReader(pdf_path)

for page in reader.pages:
    text_content += page.extract_text()

print(text_content[:500])


TRIBUNAL DE CONTAS DO ESTADO DO RIO GRANDE DO SUL 
CONCURSO PÚBLICO PARA O PROVIMENTO DE VAGAS E A FORMAÇÃO DE CADASTRO DE RESERVA 
NOS CARGOS DE AUDITOR DE CONTROLE EXTERNO E OFICIAL DE CONTROLE EXTERNO 
EDITAL Nº 1 – TCE/RS, DE 3 DE JULHO DE 2025 
 
POR DELEGAÇÃO DO PRESIDENTE DO TRIBUNAL DE CONTAS DO ESTADO DO RIO GRANDE DO 
SUL (TCE/RS), tendo em vista o disposto  na Lei Complementar Estadual nº 10.098, de 3 de fevereiro de 
1994, e suas alterações ; na Lei Estadual nº 11.424, de 6 de janeir


## Identify Elimination Criteria

### Subtask:
Analyze the extracted text to locate and understand the specific criteria for candidate elimination as outlined in the edital.


**Reasoning**:
To identify elimination criteria, I will define a list of common keywords and then search the `text_content` for occurrences of these keywords, extracting sentences around them for analysis.



In [3]:
import re

elimination_keywords = [
    "eliminação", "eliminado", "desclassificado", "exclusão", "reprovado",
    "não será aceito", "invalidado", "indeferido", "inabilitado",
    "será considerado ausente", "nulo", "cancelado", "inapto"
]

elimination_criteria_sections = []

# Split text into sentences for easier analysis
sentences = re.split(r'(?<=[.!?]) +', text_content)

for sentence in sentences:
    for keyword in elimination_keywords:
        if re.search(r'\b' + re.escape(keyword) + r'\b', sentence, re.IGNORECASE):
            elimination_criteria_sections.append(sentence.strip())
            break # Move to the next sentence once a keyword is found

# Print the first few identified sentences to review
for i, section in enumerate(elimination_criteria_sections[:20]):
    print(f"--- Found Section {i+1} ---")
    print(section)
    print("\n")

print(f"Total sections found: {len(elimination_criteria_sections)}")

--- Found Section 1 ---
5.1.6 DA AVALIAÇÃO BIOPSICOSSOCIAL 
5.1.6.1 O candidato com a inscrição deferida para concorrer como pessoa com deficiência, se não 
eliminado no concurso, será convocado para se submeter à avaliação biopsicossocial oficial promovida 
por comissão especial, composta por equipe multiprofissional formada por, no mínimo, três profissionais 
da Área Técnica do Serviço de Perícias Médicas do TCE/RS e um servidor efetivo ocupante do mesmo 
cargo pelo qual concorre o candidato , a qual analisará a qualificação do candidato como pessoa com  
deficiência, nos termos do artigo 10 da Resolução TCERS nº 898/2010, da Lei Estadual nº 13.320/2009; do 
§ 1º do artigo 2º da Lei Federal nº 13.146/2015, e suas alterações; dos artigos 3º e 4º do Decreto Federal 
nº 3.298/1999; do § 1º do artigo 1º da Lei Federal nº 12.764/2012; da Lei Federal nº  14.126/2021; e da 
Lei Federal nº 14.768/2023.


--- Found Section 2 ---
5.1.6.8.2 Se a comissão especial, após a avaliação biopsicossoci

```markdown
### Summarized Elimination Criteria:
Based on the extracted text, the following are the key elimination criteria:

1.  **Incompatibility with Disability**: If a candidate with a disability is found, after biopsychosocial evaluation, to have a disability incompatible with the exercise of the duties of the desired position, they will be eliminated (Section 5.1.6.8.2).
2.  **Refusal to be Filmed/Photographed or False Declaration**: Candidates who refuse to be filmed or photographed, or who provide false declarations, will be eliminated (Section 5.2.2.6).
3.  **False Information in Exemption Request**: Providing false information in the exemption request or using false documents will lead to elimination (Section 6.4.8.7).
4.  **Specialized Attendance (Additional Time) Mismatch**: A candidate who receives specialized attendance (additional time) but is not considered a person with a disability in the biopsychosocial evaluation will be eliminated for non-compliance with sub-item 16.2 (Section 6.4.9.2.1).
5.  **Failure to Request Specialized Attendance for Metal Detector**: Candidates who do not request specialized attendance and whose metal detector alarms indicate metallic/electronic objects that cannot be identified by manual inspection will be eliminated (Section 6.4.9.8.2).
6.  **Failure to Request Specialized Attendance for Weapons**: Candidates covered by Federal Law nº 10.826/2003 who do not request specialized attendance and carry weapons will be automatically eliminated (Section 6.4.9.9.1).
7.  **Lack of Original Identity Document**: Failure to present an original identity document as defined in sub-item 8.2 will result in automatic elimination (Section 8.3).
8.  **Non-compliance with Sub-item 8.7**: Non-observance of sub-item 8.7 will lead to non-correction of tests and subsequent elimination (Section 8.7.1).
9.  **Absence from Exams**: Non-appearance at the exam location on the scheduled days and times will result in automatic elimination (Section 8.12).
10. **Electronic Devices during Exams**: Failure to properly store (switched off in provided bag) cell phones and other electronic equipment before entering the exam room (Section 8.14.2) or if any electronic equipment turns on during the exam (Section 8.15).
11. **Academic Misconduct/Fraud during Exams**: This includes:
    *   Giving or receiving aid (8.16 a).
    *   Using unauthorized materials (books, calculators, notes, etc.) or communicating with other candidates (8.16 b).
    *   Possessing electronic devices or other prohibited objects (8.16 c).
    *   Disrespecting staff or other candidates (8.16 d).
    *   Making unauthorized notes (8.16 e).
    *   Not handing in exam materials (8.16 f).
    *   Leaving the room without supervision (8.16 g).
    *   Leaving the room with response sheets or definitive text documents (8.16 h).
    *   Disregarding instructions (8.16 i).
    *   Disrupting order (8.16 j).
    *   Using or attempting to use fraudulent means (8.16 k).
    *   Refusing to provide signature (8.16 l).
    *   Possessing non-transparent pens or unauthorized notes (8.16 m, n).
    *   Possessing weapons without authorized special assistance (8.16 o).
    *   Refusing or failing a metal detector test where the object cannot be inspected (8.16 p, 8.16.1).
    *   Refusing to transcribe a phrase for graphological examination (8.16 q).
    *   Identifying marks on evaluation documents (8.16 r).
    *   Refusing biometric data collection (8.16 s).
12. **Illicit Process/Fraud Detection**: If an illicit process is detected at any time (electronic, statistical, visual, graphological, or police investigation), the candidate's tests will be annulled, and they will be automatically eliminated (Section 8.20, 8.21).
13. **Minimum Score Requirements**: Failure to meet the minimum scores in objective tests for either Auditor de Controle Externo (ACE) or Oficial de Controle Externo (OCE) positions (Sections 9.11.4 and 9.11.5).

*   **ACE Minimum Scores (Section 9.11.4)**:
    *   P1 (General Knowledge): Less than 16.00 points.
    *   P2 (Specific Knowledge): Less than 30.00 points.
    *   Total Objective Tests: Less than 54.00 points.

*   **OCE Minimum Scores (Section 9.11.5)**:
    *   P1 (General Knowledge): Less than 10.00 points.
    *   P2 (Specific Knowledge): Less than 15.00 points.
    *   Total Objective Tests: Less than 30.00 points.

These criteria cover various stages of the selection process, from application to exam execution and final scoring.
```


## Gather Candidate Data (Required External Step)

### Subtask:
Explain that a separate list or dataset of candidates, including relevant performance/status information, is required to apply the identified elimination criteria.


To proceed with applying the elimination criteria identified from the edital and to count the non-eliminated candidates, a dataset containing comprehensive candidate information is absolutely essential. This dataset needs to include specific details for each candidate that directly correspond to the elimination rules.

For instance, based on the extracted criteria, the dataset should include, but not be limited to, the following information:

*   **Candidate ID**: A unique identifier for each candidate.
*   **P1 Score**: The score obtained in the objective test P1.
*   **P2 Score**: The score obtained in the objective test P2.
*   **Disability Status**: Information indicating if the candidate is a person with a disability and if their condition is compatible with the role.
*   **Special Request - Additional Time**: Whether the candidate requested and was granted additional time, and their disability status.
*   **Special Request - Weapon Permit**: Whether the candidate requested and was granted permission to carry a weapon.
*   **Attended Exam**: A boolean or status indicating if the candidate attended the exam.
*   **ID Document Presented**: Whether the candidate presented a valid original ID document.
*   **Used Electronic Device**: Whether the candidate was found using an unauthorized electronic device.
*   **Provided False Information**: Indication of whether the candidate provided false information during the application or exemption process.
*   **Refused Filming/Photograph**: Whether the candidate refused to be filmed or photographed.
*   **Engaged in Fraud**: Any indication of fraudulent activity during the exam.
*   **Behavior During Exam**: Information regarding the candidate's behavior, such as disrespect, disturbing order, or not following instructions.

This crucial candidate data is **not available in the current environment** and must be provided externally before the elimination criteria can be applied and the number of non-eliminated candidates can be determined. Once this data is available, we can filter out candidates based on each criterion.

An example of what such a dataset might look like in a tabular format (e.g., a CSV file or a Pandas DataFrame) is shown below:

| Candidate ID | P1 Score | P2 Score | Disability Status | Special Request - Additional Time | Special Request - Weapon Permit | Attended Exam | ID Document Presented | Used Electronic Device | Provided False Information | Refused Filming/Photograph | Engaged in Fraud | Behavior During Exam |
|--------------|----------|----------|-------------------|-----------------------------------|---------------------------------|---------------|-----------------------|------------------------|----------------------------|----------------------------|------------------|----------------------|
| 1001         | 25       | 40       | No                | No                                | No                              | Yes           | Yes                   | No                     | No                         | No                         | No               | Normal               |
| 1002         | 10       | 35       | Yes               | Yes                               | No                              | Yes           | Yes                   | No                     | No                         | No                         | No               | Normal               |
| 1003         | 30       | 50       | No                | No                                | No                              | No            | Yes                   | No                     | No                         | No                         | No               | Normal               |
| 1004         | 20       | 10       | No                | No                                | No                              | Yes           | Yes                   | No                     | No                         | No                         | No               | Normal               |
| 1005         | 15       | 20       | Yes               | Yes                               | No                              | Yes           | Yes                   | No                     | Yes                        | No                         | No               | Normal               |

Each row would represent a candidate, and each column would contain a relevant piece of information needed to check against the elimination criteria.

## Apply Criteria and Count Non-Eliminated Candidates

### Subtask:
Develop code to iterate through candidates, apply the elimination rules, and count those who are not eliminated, assuming candidate data becomes available.


## Summary:

### Q&A
The elimination criteria from the `Ed_1_TCE_RS_24_abertura.pdf` file have been successfully extracted and summarized into 13 distinct categories. Candidate data is indeed needed to apply these criteria and count non-eliminated candidates, as the current environment lacks this specific candidate information.

### Data Analysis Key Findings
*   All textual content from the `Ed_1_TCE_RS_24_abertura.pdf` file was successfully extracted and is available for analysis.
*   A comprehensive set of 13 distinct candidate elimination criteria was identified and summarized from the extracted text, covering various stages of the selection process from application to exam execution and final scoring.
*   The minimum score requirements for elimination were identified for both Auditor de Controle Externo (ACE) and Oficial de Controle Externo (OCE) positions. For ACE, candidates scoring less than 16.00 in P1, 30.00 in P2, or a total of 54.00 in objective tests will be eliminated. For OCE, the thresholds are 10.00 in P1, 15.00 in P2, or a total of 30.00.
*   To apply these elimination criteria, specific candidate data is required, including P1/P2 scores, disability status, special attendance requests, exam attendance, ID presentation, electronic device usage, false information provision, filming refusal, fraud involvement, and behavior during the exam.

### Insights or Next Steps
*   To proceed with determining non-eliminated candidates, the required comprehensive candidate dataset must be provided externally.
*   Once candidate data is available, a script can be developed to programmatically apply the 13 identified elimination criteria to each candidate entry.


# Task
Extract all textual content from the `amapla concorrencia.pdf` file to prepare it for parsing candidate data.

## Extract Text from amapla concorrencia.pdf

### Subtask:
Extract all textual content from the provided PDF file `amapla concorrencia.pdf` to make it searchable and parsable.


**Reasoning**:
To extract text from the new PDF file, I will use the `pypdf` library, iterate through each page, and concatenate the text content, then print the first 500 characters for verification.



In [4]:
from pypdf import PdfReader

amapla_pdf_path = '/content/amapla concorrencia.pdf'

amapla_text_content = ''
reader = PdfReader(amapla_pdf_path)

for page in reader.pages:
    amapla_text_content += page.extract_text()

print(amapla_text_content[:500])

1.1.5 CARGO 5: OFICIAL DE CONTROLE EXTERNO (OCE) – ESPECIALIDADE: OFICIAL 
INSTRUTIVO 10011671, Aaron Schott, 28.00, 35, 14.00, 26, 42.00 / 10015729, Abelmaim 
Alessandro da Silva, 13.00, 28, 19.00, 33, 32.00 / 10011611, Abimael Gomes Leite, 0.00, 25, 18.00, 
34, 18.00 / 10001725, Adaiane Ribeiro de Sousa, 22.00, 26, 15.00, 22, 37.00 / Porto Alegre, quinta-
feira, 27 de novembro de 2025 Diário Oficial Nº 232 157 10006504, Adailson de Lima Rodrigues, 
14.00, 32, 10.00, 30, 24.00 / 10013254, Adair


**Reasoning**:
The extracted text `amapla_text_content` contains candidate information in a semi-structured format. I need to parse this text using regular expressions to extract individual candidate data points like ID, name, P1 score, P2 score, and total score, and then store them in a pandas DataFrame for further processing against elimination criteria.



In [5]:
import re
import pandas as pd

candidate_data = []

# The pattern looks for:
# 1. Candidate ID (digits)
# 2. Candidate Name (text with spaces, potentially accented characters)
# 3. P1 Score (float)
# 4. An intermediate number (likely questions answered, ignoring for now)
# 5. P2 Score (float)
# 6. Another intermediate number (ignoring for now)
# 7. Total Score (float)

# Updated pattern to capture the P1 Score, the number of questions for P1, P2 Score, the number of questions for P2, and the Total Score.
# The regex has been updated based on the format 'ID, Name, P1_Score, P1_Questions, P2_Score, P2_Questions, Total_Score'
# Example: 10011671, Aaron Schott, 28.00, 35, 14.00, 26, 42.00

pattern = re.compile(r'(\d+),\s*([^,]+),\s*(\d+\.?\d*),\s*(\d+),\s*(\d+\.?\d*),\s*(\d+),\s*(\d+\.?\d*)')

# Split the text by '/' to get individual candidate entries, as observed in the sample output
candidate_entries = amapla_text_content.split(' / ')

for entry in candidate_entries:
    match = pattern.search(entry)
    if match:
        candidate_id = match.group(1)
        candidate_name = match.group(2).strip()
        p1_score = float(match.group(3))
        p1_questions = int(match.group(4))
        p2_score = float(match.group(5))
        p2_questions = int(match.group(6))
        total_score = float(match.group(7))

        candidate_data.append({
            'Candidate ID': candidate_id,
            'Candidate Name': candidate_name,
            'P1 Score': p1_score,
            'P1 Questions': p1_questions,
            'P2 Score': p2_score,
            'P2 Questions': p2_questions,
            'Total Score': total_score
        })

candidates_df = pd.DataFrame(candidate_data)

print(f"Extracted {len(candidates_df)} candidates. First 5 rows:")
print(candidates_df.head())


Extracted 5549 candidates. First 5 rows:
  Candidate ID                  Candidate Name  P1 Score  P1 Questions  \
0     10011671                    Aaron Schott      28.0            35   
1     10015729  Abelmaim \nAlessandro da Silva      13.0            28   
2     10011611             Abimael Gomes Leite       0.0            25   
3     10001725        Adaiane Ribeiro de Sousa      22.0            26   
4     10006504      Adailson de Lima Rodrigues      14.0            32   

   P2 Score  P2 Questions  Total Score  
0      14.0            26         42.0  
1      19.0            33         32.0  
2      18.0            34         18.0  
3      15.0            22         37.0  
4      10.0            30         24.0  


**Reasoning**:
Based on the extracted candidate data and the identified elimination criteria, the next logical step is to apply the minimum score requirements for the 'Oficial de Controle Externo (OCE)' position, as the provided PDF (amapla concorrencia.pdf) clearly indicates candidates for this role. I will add a column to mark eliminated candidates based on these criteria and display the count of non-eliminated candidates.



In [6]:
import numpy as np

# Initialize an 'Eliminated' column to False for all candidates
candidates_df['Eliminated'] = False
candidates_df['Elimination Reason'] = ''

# Define OCE minimum score criteria from Section 9.11.5
OCE_MIN_P1 = 10.00
OCE_MIN_P2 = 15.00
OCE_MIN_TOTAL = 30.00

# Apply OCE score elimination criteria
# A candidate is eliminated if: (P1 < 10) OR (P2 < 15) OR (Total < 30)
score_elimination_condition = (
    (candidates_df['P1 Score'] < OCE_MIN_P1) |
    (candidates_df['P2 Score'] < OCE_MIN_P2) |
    (candidates_df['Total Score'] < OCE_MIN_TOTAL)
)

# Mark candidates as eliminated based on score criteria
candidates_df.loc[score_elimination_condition, 'Eliminated'] = True
candidates_df.loc[score_elimination_condition, 'Elimination Reason'] = \
    candidates_df.loc[score_elimination_condition, 'Elimination Reason'].apply(lambda x: (x + '; ') if x else '') + 'Failed minimum score requirements (OCE)'


# Count non-eliminated candidates
non_eliminated_count = candidates_df[~candidates_df['Eliminated']].shape[0]
eliminated_count = candidates_df[candidates_df['Eliminated']].shape[0]

print(f"Total candidates: {len(candidates_df)}")
print(f"Candidates eliminated based on OCE score criteria: {eliminated_count}")
print(f"Candidates NOT eliminated based on OCE score criteria: {non_eliminated_count}")

print("\nFirst 10 candidates eliminated by score:")
print(candidates_df[candidates_df['Eliminated']].head(10)[['Candidate ID', 'Candidate Name', 'P1 Score', 'P2 Score', 'Total Score', 'Elimination Reason']])


Total candidates: 5549
Candidates eliminated based on OCE score criteria: 2819
Candidates NOT eliminated based on OCE score criteria: 2730

First 10 candidates eliminated by score:
   Candidate ID                        Candidate Name  P1 Score  P2 Score  \
0      10011671                          Aaron Schott      28.0      14.0   
2      10011611                   Abimael Gomes Leite       0.0      18.0   
4      10006504            Adailson de Lima Rodrigues      14.0      10.0   
5      10013254                  Adair da Rosa Savois       7.0      29.0   
7      10019809         Adao Mariano Vaz \ndos Santos       6.0       5.0   
9      10019778               Aderson Pereira Machado       7.0       8.0   
13     10002850  Adoniran Josue \nde Carvalho Avelino      10.0      12.0   
14     10002896                 Adrian da Cruz Santos      11.0      10.0   
15     10015391             Adrian Domingues de Borba      10.0      12.0   
16     10015174             Adrian Galecki Przibi

## Sort and Format Non-Eliminated Candidates for Excel

### Subtask:
Filter the candidates to include only those who were 'Not Eliminated', sort them in descending order based on their total objective test scores, and format the output in a structured way suitable for direct import into Excel.


**Reasoning**:
I will filter the `candidates_df` to keep only non-eliminated candidates, sort them by total score, display the top 10, and then save the results to a CSV file suitable for Excel.



In [7]:
non_eliminated_candidates_df = candidates_df[~candidates_df['Eliminated']].copy()

# Sort by 'Total Score' in descending order
non_eliminated_candidates_df.sort_values(by='Total Score', ascending=False, inplace=True)

print("First 10 non-eliminated candidates, sorted by Total Score:")
print(non_eliminated_candidates_df.head(10)[['Candidate ID', 'Candidate Name', 'P1 Score', 'P2 Score', 'Total Score']])

# Save to CSV for Excel import
output_csv_path = 'non_eliminated_oce_candidates.csv'
non_eliminated_candidates_df.to_csv(output_csv_path, index=False)

print(f"\nNon-eliminated candidates saved to '{output_csv_path}'")

First 10 non-eliminated candidates, sorted by Total Score:
     Candidate ID                  Candidate Name  P1 Score  P2 Score  \
552      10012249              Augusto Furlanetto      44.0      47.0   
3181     10005669  Laura Beatriz de Freitas Silva      42.0      48.0   
4090     10002544    Mauricio Tavares Bosquerolli      42.0      44.0   
1826     10010875                     Felipe Leme      38.0      47.0   
2367     10011275            Heitor Krug Einsfeld      40.0      44.0   
5206     10013363         Thiago Francesco Borges      40.0      43.0   
2941     10016538     Julia Rosario Frank \nNejar      35.0      47.0   
4297     10007489           Nicolas Farias Kruger      42.0      40.0   
1491     10005849         Eduardo Demaria Filippi      42.0      40.0   
2958     10001611              Juliana Both Engel      41.0      41.0   

      Total Score  
552          91.0  
3181         90.0  
4090         86.0  
1826         85.0  
2367         84.0  
5206         83.0

## Final Task

### Subtask:
Provide the final list of non-eliminated candidates for the 'Oficial de Controle Externo' position, sorted by objective scores, in a format suitable for Excel.


## Summary:

### Q&A
The final list of non-eliminated candidates for the 'Oficial de Controle Externo' position, sorted by objective scores, has been provided in a CSV file named `non_eliminated_oce_candidates.csv`, formatted for Excel.

### Data Analysis Key Findings
*   Textual content from `amapla concorrencia.pdf` was successfully extracted, containing structured candidate data.
*   A total of 5549 candidate entries were parsed into a DataFrame, including Candidate ID, Name, P1 Score, P1 Questions, P2 Score, P2 Questions, and Total Score.
*   Minimum score requirements for the 'Oficial de Controle Externo' position (P1 Score < 10.00, P2 Score < 15.00, or Total Score < 30.00) were applied to the candidates.
*   Out of 5549 candidates, 2819 were eliminated based on these score criteria, resulting in 2730 non-eliminated candidates.
*   The non-eliminated candidates were sorted in descending order by their 'Total Score'. The highest-scoring non-eliminated candidate, Augusto Furlanetto, achieved a Total Score of 91.0.
*   The final list of non-eliminated candidates was exported to a CSV file named `non_eliminated_oce_candidates.csv`, suitable for direct import into Excel.

### Insights or Next Steps
*   The generation of a structured CSV file allows for straightforward distribution and further manual review or processing of the qualified candidates by hiring managers.
*   Consider implementing additional qualification criteria beyond objective scores, such as qualitative assessments or other screening phases, to further refine the candidate pool.


In [9]:
print(non_eliminated_candidates_df.head(2000)[['Candidate ID', 'Candidate Name', 'P1 Score', 'P2 Score', 'Total Score']])

     Candidate ID                        Candidate Name  P1 Score  P2 Score  \
552      10012249                    Augusto Furlanetto      44.0      47.0   
3181     10005669        Laura Beatriz de Freitas Silva      42.0      48.0   
4090     10002544          Mauricio Tavares Bosquerolli      42.0      44.0   
1826     10010875                           Felipe Leme      38.0      47.0   
2367     10011275                  Heitor Krug Einsfeld      40.0      44.0   
...           ...                                   ...       ...       ...   
5278     10005279                  Valmir de Mello Boff      22.0      18.0   
467      10009299                 Annelise Mendes Greff      20.0      20.0   
466      10001003           Annelise \nBastos Guerreiro      19.0      21.0   
5247     10012315  Tiago Pinheiro dos \nSantos de Souza      24.0      16.0   
3535     10017102                  Luciane Ferret Flach      20.0      20.0   

      Total Score  
552          91.0  
3181       

In [11]:
import pandas as pd

# Path to the previously saved CSV file
output_csv_path = 'non_eliminated_oce_candidates.csv'

# Read the CSV file into a DataFrame
csv_df = pd.read_csv(output_csv_path)

# Display the head of the DataFrame to show names and scores in a table format
print("Conteúdo do arquivo CSV de candidatos não eliminados (primeiras 10 linhas):")
print(csv_df.head(2000)[['Candidate ID', 'Candidate Name', 'P1 Score', 'P2 Score', 'Total Score']])

Conteúdo do arquivo CSV de candidatos não eliminados (primeiras 10 linhas):
      Candidate ID                        Candidate Name  P1 Score  P2 Score  \
0         10012249                    Augusto Furlanetto      44.0      47.0   
1         10005669        Laura Beatriz de Freitas Silva      42.0      48.0   
2         10002544          Mauricio Tavares Bosquerolli      42.0      44.0   
3         10010875                           Felipe Leme      38.0      47.0   
4         10011275                  Heitor Krug Einsfeld      40.0      44.0   
...            ...                                   ...       ...       ...   
1995      10005279                  Valmir de Mello Boff      22.0      18.0   
1996      10009299                 Annelise Mendes Greff      20.0      20.0   
1997      10001003           Annelise \nBastos Guerreiro      19.0      21.0   
1998      10012315  Tiago Pinheiro dos \nSantos de Souza      24.0      16.0   
1999      10017102                  Luciane 

In [12]:
page_size = 50 # You can change this to 100 if preferred

total_candidates = len(csv_df)
total_pages = (total_candidates + page_size - 1) // page_size

for i in range(total_pages):
    start_index = i * page_size
    end_index = min((i + 1) * page_size, total_candidates)

    print(f"\n--- Página {i+1} de {total_pages} (Candidatos {start_index + 1} a {end_index}) ---")
    print(csv_df.iloc[start_index:end_index][['Candidate ID', 'Candidate Name', 'P1 Score', 'P2 Score', 'Total Score']].to_string(index=False))



--- Página 1 de 55 (Candidatos 1 a 50) ---
 Candidate ID                      Candidate Name  P1 Score  P2 Score  Total Score
     10012249                  Augusto Furlanetto      44.0      47.0         91.0
     10005669      Laura Beatriz de Freitas Silva      42.0      48.0         90.0
     10002544        Mauricio Tavares Bosquerolli      42.0      44.0         86.0
     10010875                         Felipe Leme      38.0      47.0         85.0
     10011275                Heitor Krug Einsfeld      40.0      44.0         84.0
     10013363             Thiago Francesco Borges      40.0      43.0         83.0
     10016538         Julia Rosario Frank \nNejar      35.0      47.0         82.0
     10007489               Nicolas Farias Kruger      42.0      40.0         82.0
     10005849             Eduardo Demaria Filippi      42.0      40.0         82.0
     10001611                  Juliana Both Engel      41.0      41.0         82.0
     10011167                Lucas Castro P

In [14]:
import numpy as np

# The problematic name was in cell ltL12oNvdfuu, which was generated in the last turn
# Let's adjust the search string to match the format observed in the dataframe, or use a regex.
# A more robust way is to strip whitespace/newlines from the 'Candidate Name' column for comparison if it's inconsistent.

# First, let's clean the 'Candidate Name' column in candidates_df to handle potential newlines for consistent searching.
candidates_df_cleaned_names = candidates_df.copy()
candidates_df_cleaned_names['Candidate Name'] = candidates_df_cleaned_names['Candidate Name'].str.replace('\n', ' ', regex=False).str.strip()

# Now search using the cleaned name and check for the candidate's presence and details.
candidate_name_search_cleaned = "Vitor de Brito Cardoso Oliveira"
vitor_candidate_details = candidates_df_cleaned_names[candidates_df_cleaned_names['Candidate Name'].str.contains(candidate_name_search_cleaned, na=False, case=False)]

if not vitor_candidate_details.empty:
    print("Detalhes do candidato Vitor de Brito Cardoso Oliveira (com nome corrigido):")
    display(vitor_candidate_details[['Candidate ID', 'Candidate Name', 'P1 Score', 'P2 Score', 'Total Score', 'Eliminated', 'Elimination Reason']])

    # Check if the candidate is in the non_eliminated_candidates_df
    vitor_non_eliminated = non_eliminated_candidates_df[non_eliminated_candidates_df['Candidate ID'] == vitor_candidate_details['Candidate ID'].iloc[0]]

    if not vitor_non_eliminated.empty:
        # Find the rank of the candidate among non-eliminated candidates
        # The non_eliminated_candidates_df is already sorted by 'Total Score' in descending order
        rank = non_eliminated_candidates_df.index.get_loc(vitor_non_eliminated.index[0]) + 1
        print(f"\nO candidato Vitor de Brito Cardoso Oliveira está classificado na posição: {rank} (de {len(non_eliminated_candidates_df)} candidatos não eliminados).")
    else:
        print("\nO candidato Vitor de Brito Cardoso Oliveira foi eliminado com base nos critérios.")
else:
    print(f"Candidato '{candidate_name_search_cleaned}' não encontrado no DataFrame, mesmo após a correção de nova linha.")


Candidato 'Vitor de Brito Cardoso Oliveira' não encontrado no DataFrame, mesmo após a correção de nova linha.
